atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -660,7 +661,7 @@ private Map<Cell, Value> getAllCellsForRows(
             final CassandraServer host, final TableReference tableRef, final List<byte[]> rows, final long startTs)
             throws Exception {
 
-        final ListMultimap<ByteBuffer, ColumnOrSuperColumn> result = LinkedListMultimap.create();
+        ListMultimap<ByteBuffer, ColumnOrSuperColumn> result = ArrayListMultimap.create();
```

Comment:
Do we have any metrics on key to value cardinality here? I'm wondering if we'll bloat with the default ArrayList capacity of 10 entries once we add a single element, in which case we might want to provide a value collection factory.

Reply:
Good call, I think we can probably just use the `ArrayListMultimap.create` method that takes an `expectedKeys` and `expectedValues` and use 1 for `expectedValues`:
- We're very likely to have one value (client don't often query for keys that don't exist)
- It avoids copying in the single value case, which is fairly common
- We add the values for a given key all at once, so this should require additional growth because the initial capacity is small